### PR TITLE
[CCXDEV-12174] Reorganize requirement files

### PR DESCRIPTION
**Pull request by epapbak**
# Description

Behave imports everything defined under the `features/steps` directory, forcing to always install all the requirements independently of the tests being run.

Fixes CCXDEV-12174

## Type of change

- Refactor (refactoring code, removing useless files)

## Testing steps

- Executed each feature

## Checklist
* [ ] Pylint passes for Python sources
* [ ] sources has been pre-processed by Black
* [ ] updated documentation wherever necessary
* [ ] new tests can be executed both locally and within docker container
* [ ] new tests have been included in scenario list (make update-scenarios)

### CODE REVIEW

File: insights_sha_extractor_test.sh [hunk 1]
```diff
@@ -32,13 +32,12 @@ function clone_service() {
...
 function install_reqs() {
     pip install -r requirements.txt
-    pip install -r requirements/insights_sha_extractor.txt
 }
 
 function prepare_venv() {
     echo "Preparing environment"
     # shellcheck disable=SC1091
-    virtualenv -p python3 venv && source venv/bin/activate
+    virtualenv -p python3 venv && source venv/bin/activate && install reqs
```
Comment (juandspy) [inline insights_sha_extractor_test.sh:40]: so if I use NOENV it won't install anything?

Comment (epapbak) [reply insights_sha_extractor_test.sh:40]: yep. I'm thinking of removing that logic too, but thought on doing that on another PR.


Should I clean it up too?

Comment (epapbak) [reply insights_sha_extractor_test.sh:40]: Clean up would be something like "always try to create the virtual env and always try to install the requirements"

File: ccx_upgrade_risk_data_eng_tests.sh [hunk 2]
```diff
@@ -22,14 +22,10 @@ PATH_TO_LOCAL_DATA_ENG_SERVICE=${PATH_TO_LOCAL_DATA_ENG_SERVICE:="../ccx-upgrade
...
 
-function install_reqs() {
-    pip install -r requirements.txt
-}
-
 function prepare_venv() {
     echo "Preparing environment"
     # shellcheck disable=SC1091
-    virtualenv -p python3 venv && source venv/bin/activate && pip install -r requirements/upgrades_data_eng_service.txt
+    virtualenv -p python3 venv && source venv/bin/activate && pip install -r requirements.txt
```
Comment (ikerreyes) [inline ccx_upgrade_risk_data_eng_tests.sh:28]: If they are installed a Dockerfile level, do we still need this? Why not setting the venv in the dockerfile?

File: smart_proxy_tests.sh [hunk 3]
```diff
@@ -25,7 +25,7 @@ function install_reqs() {
 function prepare_venv() {
     echo "Preparing environment"
     # shellcheck disable=SC1091
-    virtualenv -p python3 venv && source venv/bin/activate && pip install -r requirements/insights_results_aggregator_mock.txt || exit 1
+    virtualenv -p python3 venv && source venv/bin/activate && install_reqs || exit 1
```
Comment (ikerreyes) [inline smart_proxy_tests.sh:28]: some funcionts use `install_reqs` and others don't. Can we standardize?

File: insights_sha_extractor_test.sh (hunk 1 above)
Comment (juandspy) [reply insights_sha_extractor_test.sh:40]: sorry, I missed this comment

> "always try to create the virtual env and always try to install the requirements"

yes, I agree

Comment (juandspy) [reply insights_sha_extractor_test.sh:40]: but feel free to do it separatedly

Comment (epapbak) [reply insights_sha_extractor_test.sh:40]: I added the changes in the previous patch

File: smart_proxy_tests.sh (hunk 3 above)
Comment (epapbak) [reply smart_proxy_tests.sh]: this should be solved

**Review by matysek — APPROVED** [review]

**Review by ikerreyes — APPROVED** [review]

**Pull request merged**